mlflow/tracking/artifact_utils.py

```diff
@@ -53,14 +58,40 @@ def _download_artifact_from_uri(artifact_uri, output_path=None):
     :param output_path: The local filesystem path to which to download the artifact. If unspecified,
                         a local output path will be created.
     """
+    artifact_repo_type = get_artifact_repository_type(artifact_uri)
+
+    if artifact_repo_type == ArtifactRepositoryType.FileSystem:
+        parsed_uri = urllib.parse.urlparse(artifact_uri)
+        prefix = ""
+        if parsed_uri.scheme and not parsed_uri.path.startswith("/"):
+            # relative path is a special case, urllib does not reconstruct it properly
+            prefix = parsed_uri.scheme + ":"
+            parsed_uri = parsed_uri._replace(scheme="")
+        artifact_path = posixpath.basename(parsed_uri.path)
+        parsed_uri = parsed_uri._replace(path=posixpath.dirname(parsed_uri.path))
+        root_uri = prefix + urllib.parse.urlunparse(parsed_uri)
+        return get_artifact_repository(artifact_uri=root_uri).download_artifacts(
+            artifact_path=artifact_path, dst_path=output_path)
+    else:
+        db_uri, artifact_path = extract_db_uri_and_artifact_path_from_uri(artifact_uri)
+        return get_artifact_repository(artifact_uri=db_uri).download_artifacts(
+            artifact_path=artifact_path, dst_path=output_path)
+
+
+def extract_db_uri_and_artifact_path_from_uri(artifact_uri):
```

Comment:
Do we need a new db uri extraction method here, or can we move the logic into `extract_db_uri_and_path`? Why do we parse additional information about the URI scheme here but not in `extract_db_uri_and_path`?